### PR TITLE
feat(exporter): add Markdown table export

### PR DESCRIPTION
**Pull request by eric-song-dev**
Changes proposed in this pull request:

- Add a new `MarkdownExporter` for exporting the current project view as a GitHub-flavored Markdown table.
- Register the `md` export format in the exporter registry and add it to the Export menu as `Markdown table`.
- Add unit tests covering normal table output, empty cells, escaped pipe characters, and newline handling inside cells.

Tests run:

- `mvn -pl main -Dtest=MarkdownExporterTests test`

### CODE REVIEW

**Comment by magdmartin** [comment]
@eric-song-dev  thanks for your interest in OpenRefine. Is your change related to an existing issue?  If so, can you link it? 

I am curious regarding the use case for adding a markdown table export.

File: main/src/com/google/refine/exporters/MarkdownExporter.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+                    writer.write("\n");
+
+                    if (isHeader) {
+                        writer.write("|");
+                        for (int i = 0; i < cells.size(); i++) {
+                            writer.write(" --- |");
+                        }
+                        writer.write("\n");
+                    }
+                } catch (IOException e) {
```
Comment (durkagiri21-ship-it) [inline main/src/com/google/refine/exporters/MarkdownExporter.java:88]: Should this IOException be logged instead of silently ignored? Swallowing exceptions here could make it hard to debug export failures in production. Might be worth adding at least a logger.error() call.

**Comment by tfmorris** [comment]
Thanks for the contribution, but closing since this doesn't address an identified need in an open, triaged issue. Unfortunately, we don't have the bandwidth to review PRs which don't address pressing issues. 

Please look for an issue which has been tagged with the "Good First Issue" tag or which looks easy to solve for you. Thanks!
